libs/ui/design-system/src/lib/checkbox.tsx

```diff
@@ -1,17 +1,32 @@
-import { Checkbox as CheckboxPrimitive } from '@nmit-coursition/ui/primitives/checkbox'
+import * as CheckboxPrimitive from '@nmit-coursition/ui/primitives/checkbox'
 import { Label } from '@nmit-coursition/ui/primitives/label'
+import { cn } from '@nmit-coursition/ui/utils'
+import { type VariantProps, cva } from 'class-variance-authority'
 
-interface CheckboxProps {
+const checkboxVariants = cva('', {
+  variants: {
+    variant: {
+      outlined: '',
+      critical: '', // applies a red border
```

Comment:
**suggestion:** Critical is not a status, **red** colour is applied when checkbox is invalid (e.g. it has to be checked like _"I agree with Terms & Conditions."_).